modules/charon-core/src/test/java/org/wso2/charon3/core/protocol/endpoints/MeResourceManagerTest.java

```diff
@@ -0,0 +1,1549 @@
+
+    }
+
+    @DataProvider(name = "dataForUpdateWithPATCHNotImplementedException")
+    public Object[][] dataToUpdateWithPATCHNotImplementedException()
+            throws BadRequestException, CharonException, InternalErrorException {
+
+        SCIMResourceTypeSchema schema = SCIMResourceSchemaManager.getInstance().getUserResourceSchema();
+        JSONDecoder decoder = new JSONDecoder();
+
+        User userOld = decoder.decodeResource(NEWUSER_SCIM_OBJECT_STRING_FOR_PATCH, schema, new User());
+        String id = userOld.getId();
+
+        return new Object[][]{
+                {id, NEW_USER_SCIM_OBJECT_STRING_FOR_PATCH_UPDATE, "userName", null,
+                        ResponseCodeConstants.CODE_NOT_IMPLEMENTED}
+        };
+    }
+
+    //NotImplementedException
+    @Test(dataProvider = "dataForUpdateWithPATCHNotImplementedException")
+    public void testUpdateWithPATCHNotImplementedException(String existingId, String scimObjectString,
+                                                           String attributes, String excludeAttributes,
+                                                           int expectedScimResponseStatus)
+            throws BadRequestException, NotFoundException, CharonException {
+
+        SCIMResourceTypeSchema schema = SCIMResourceSchemaManager.getInstance().getUserResourceSchema();
+
+        mockStatic(AbstractResourceManager.class);
+
+        when(AbstractResourceManager.getResourceEndpointURL(SCIMConstants.USER_ENDPOINT))
+                .thenReturn(ENDPOINT_ME);
+        when(AbstractResourceManager.getEncoder()).thenReturn(new JSONEncoder());
+        when(AbstractResourceManager.getDecoder()).thenReturn(new JSONDecoder());
+        when(AbstractResourceManager.encodeSCIMException(any(NotImplementedException.class)))
+                .thenReturn(getEncodeSCIMExceptionObject(new NotImplementedException()));
+
+        when(userManager.getMe(existingId,
+                ResourceManagerUtil.getAllAttributeURIs(schema))).thenThrow(NotImplementedException.class);
+        SCIMResponse outputScimResponse = meResourceManager.updateWithPATCH(existingId, scimObjectString,
+                userManager, attributes, excludeAttributes);
+        Assert.assertEquals(outputScimResponse.getResponseStatus(), expectedScimResponseStatus);
+
+    }
+
+    @DataProvider(name = "dataForUpdateWithPATCHInternalErrorException")
+    public Object[][] dataToUpdateWithPATCHInternalErrorException()
+            throws BadRequestException, CharonException, InternalErrorException {
+
+        SCIMResourceTypeSchema schema = SCIMResourceSchemaManager.getInstance().getUserResourceSchema();
+        JSONDecoder decoder = new JSONDecoder();
+
+        User userOld = decoder.decodeResource(NEWUSER_SCIM_OBJECT_STRING_FOR_PATCH, schema, new User());
+        String id = userOld.getId();
+
+        User userNew = decoder.decodeResource(NEW_USER_SCIM_OBJECT_STRING_FOR_PATCH_UPDATE, schema, new User());
```

Comment:
why do we need this?

Reply:
yes , its unused (Removed : 07970a9868001bf545c6d759c3ecc8b7a32498bf)